fulcio.proto

```diff
@@ -0,0 +1,123 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.fulcio.v2;
+
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/empty.proto";
+
+option go_package = "github.com/sigstore/fulcio/pkg/generated/protobuf";
+option java_package = "dev.sigstore.fulcio.v2";
+option java_multiple_files = true;
+option java_outer_classname = "FulcioProto";
+
+service CA {
+    /**
+     * Returns an X509 certificate created by the Fulcio certificate authority for the given request parameters
+     */
+    rpc CreateSigningCertificate(CreateSigningCertificateRequest) returns (SigningCertificate){
+        option (google.api.http) = {
+          post: "/api/v2/signingCert"
+          body: "*"
+        };
+    }
+    /**
+     * Returns the bundle of certificates that can be used to validate code signing certificates issued by this Fulcio instance
+     */
+    rpc GetTrustBundle(google.protobuf.Empty) returns (TrustBundle){
+        option (google.api.http) = {
+          get: "/api/v2/trustBundle"
+        };
+    }
+}
+
+message CreateSigningCertificateRequest {
+    /*
+     * Identity information about who possesses the private / public key pair presented
+     */
+    Credentials credentials    = 1 [(google.api.field_behavior) = REQUIRED];
+    /*
+     * The public key to be stored in the requested certificate
+     */
+    PublicKey public_key       = 2 [(google.api.field_behavior) = REQUIRED];
+    /*
+     * Proof that the client possesses the private key; must be verifiable by provided public key
+     *
+     * This is a currently a signature over the `sub` claim from the OIDC identity token
+     */
+    bytes proof_of_possession  = 3 [(google.api.field_behavior) = REQUIRED];
+}
+
+message Credentials {
+    oneof credentials {
+        /*
+        * The OIDC token that identifies the caller
+        */
+        string oidc_identity_token = 1;
+    }
+}
+
+message PublicKey {
+    /*
+     * The cryptographic algorithm to use with the key material
+     */
+    PublicKeyAlgorithm algorithm = 1;
+    /*
+     * PEM encoded public key
+     */
+    string content               = 2 [(google.api.field_behavior) = REQUIRED];
+}
+
+// (-- api-linter: core::0142::time-field-type=disabled
+//     aip.dev/not-precedent: SCT is defined in RFC6962 and we keep the name consistent for easier understanding. --)
+message SigningCertificate {
+    /*
+     * The signed certificate timestamp is a promise for including the certificate in
+     * a certificate transparency log. It can be "stapled" to verify the inclusion of
+     * a certificate in the log in an offline fashion.
+     */
+    bytes signed_certificate_timestamp = 1;
```

Comment:
Can we update the comment to say this is optional, as the SCT could also be embedded? Also IIRC we had chatted about a one-of, do we want to do that? Or just leave this as an optional field?